hydra/garaga/modulo_circuit.py

```diff
@@ -666,6 +700,23 @@ def fp2_square(self, X: list[ModuloCircuitElement]):
             self.double(self.mul(X[0], X[1])),
         ]
 
+    def fp2_sqrt(
+        self, element: list[ModuloCircuitElement]
+    ) -> list[ModuloCircuitElement]:
+        assert self.compilation_mode == 0, "fp2_sqrt is not supported in cairo 1 mode"
+
+        root = Fp2(element[0].felt, element[1].felt).sqrt()
+        if root is None:
+            raise ValueError("No square root found for the given element")
+
+        # ToDo: the selection of which root to select is an unvalidated hint. Not sure if this can stay like this
```

Comment:
Not sure if this is ok tbh. Probably its better to enforce his properly. Thoughts?